mdata/cwr.go

```diff
@@ -13,30 +13,48 @@ import (
 	"github.com/raintank/schema"
 )
 
+//go:generate msgp
+//msgp:ignore ChunkWriteRequest
+
 type ChunkSaveCallback func()
 
 // ChunkWriteRequest is a request to write a chunk into a store
-//go:generate msgp
 type ChunkWriteRequest struct {
-	Callback  ChunkSaveCallback `msg:"-"`
-	Key       schema.AMKey      `msg:"key,extension"`
+	ChunkWriteRequestPayload
+	Callback ChunkSaveCallback
+	Key      schema.AMKey
+}
+
+func NewChunkWriteRequest(callback ChunkSaveCallback, key schema.AMKey, ttl, t0 uint32, data []byte, ts time.Time) ChunkWriteRequest {
+	return ChunkWriteRequest{ChunkWriteRequestPayload{ttl, t0, data, ts}, callback, key}
+}
+
+// ChunkWriteRequestWithoutOrg is used by the importer utility to send cwrs over the network
```

Comment:
if you need to document a structure in mdata that it's [only] used by a specific tool, then we should just move that structure to a package for that tool.
e.g. create a package `importer` and create a `ChunkWriteRequest` type in there?

Reply:
all of cmd/mt-whisper-importer-reader/conversion.go should go into an importer package as well, cmd is meant for the main program code.

Reply:
good idea. where would you suggest i put that `importer` package? should that live in `mdata` since it is all kind of related to data handling & conversion?

Reply:
sounds good